### ISSUE TITLE
dropdown item without href attribute doesn't trigger onclick through keyboard "enter"

### ISSUE THREAD

**Issue by janett-baresel** [Esri/calcite-components#4911]
### Summary

The calcite dropdown generally works with keyboard navigation. However, it only works the dropdown item has an href is defined when clicking on the item. If the code only uses the onclick event, the keyboard "enter" key is not triggering the callback on onclick.

### Actual Behavior

When only an onclick handler is defined, pressing enter is not triggering the code path that is triggered by clicking on the drop-down item.

### Expected Behavior

When an onclick handler (and no href) is defined, I expect this to be triggered also when I press enter on the keyboard.

### Reproduction Sample

https://codepen.io/Programmine/pen/WNzGyPj?editors=1000

### Reproduction Steps

```
<calcite-dropdown-item
      active={active}
      class={classes(CSS.entry, properties.disabled && CSS.entryDisabled)}
      icon-start={properties.iconStart}
      key={properties.key}
      title={properties.title}
      onclick={properties.onClick}
    >
      {properties.label}
      {children}
</calcite-dropdown-item>
```

I can add another event handler to calcite-dropdown-item and it will work:
```
<calcite-dropdown-item
      active={active}
      class={classes(CSS.entry, properties.disabled && CSS.entryDisabled)}
      icon-start={properties.iconStart}
      key={properties.key}
      title={properties.title}
      onclick={properties.onClick}
      onkeydown={(e: KeyboardEvent) => {
        if (e.key === "Enter") {
          properties.onClick();
        }
      }}
>...
```

It seems a bit cumbersome though and I would expect that the component can do this out of the box.

https://github.com/Esri/calcite-components/blob/master/src/components/dropdown-item/dropdown-item.tsx#L210-L212

### Reproduction Version

beta.82

<sub>SVCH</sub>

**Comment by driskull** [comment]
> When an onclick handler (and no href) is defined, I expect this to be triggered also when I press enter on the keyboard.

@janett-baresel That onclick handler is only going to fire onclick. You would need to set an `onkeydown` handler as well.

**Comment by driskull** [comment]
@janett-baresel if you want to know when something is clicked or enter key, you should listen to `calciteDropdownSelect` event.

**Comment by driskull** [comment]
@jcfranco @benelan this issue may not be valid.

One part of it we could improve is that if the user supplies their own `onclick` it overrides the onclick we set on the host. We could move that `onclick` to an internal container. Should we do this?

**Comment by jcfranco** [comment]
> One part of it we could improve is that if the user supplies their own onclick it overrides the onclick we set on the host. We could move that onclick to an internal container. Should we do this?

That sounds complex. 🤯

Taking a step back, looks like that `onclick` is bubbling from the internal anchor and is available via the global HTML properties. I agree that `calciteDropdownSelect` should be used instead. @janett-baresel do you have a selection in your use case? If so, you'd have to wait for https://github.com/Esri/calcite-components/issues/1967 to land to acquire the selected element behind the selection.

**Comment by driskull** [comment]
I think we can close this issue and work on #1967. 

- The codepen should have a `<calcite-dropdown-group>` surrounding the items.
- The `calciteDropdownSelect` should be used to know when something has been selected via keyboard or click.

**Comment by driskull** [comment]
@janett-baresel closing. Let me know if you have any further questions.

**Comment by janett-baresel** [comment]
Thanks for the clarification. Will change and let you know if I run into any more issues.
